internal/services/controllers/v1/olap/controller.go

```diff
@@ -332,6 +334,59 @@ func (tc *OLAPControllerImpl) handleCreatedDAG(ctx context.Context, tenantId str
 	return tc.repo.OLAP().CreateDAGs(ctx, tenantId, createDAGOpts)
 }
 
+func (tc *OLAPControllerImpl) handleCreateEventTriggers(ctx context.Context, tenantId string, payloads [][]byte) error {
+	msgs := msgqueue.JSONConvert[tasktypes.CreatedEventTriggerPayload](payloads)
+
+	eventIds := make([]int64, 0)
+	eventInsertedAts := make([]pgtype.Timestamptz, 0)
+	runExternalIds := make([]pgtype.UUID, 0)
+	runInsertedAts := make([]pgtype.Timestamptz, 0)
+
+	for _, msg := range msgs {
+		for _, payload := range msg.Payloads {
+			opts := sqlcv1.CreateEventParams{
+				Tenantid:           sqlchelpers.UUIDFromStr(tenantId),
+				Generatedat:        sqlchelpers.TimestamptzFromTime(payload.EventGeneratedAt),
+				Key:                payload.EventKey,
+				Payload:            payload.EventPayload,
+				AdditionalMetadata: payload.EventAdditionalMetadata,
+			}
+
+			event, err := tc.repo.OLAP().CreateEvent(ctx, tenantId, opts)
```

Comment:
unsure about this - I think we might need to do it to get the id back, but it's a bit unfortunate to not be able to do this whole insert (both the events and the trigger bridge table) in one shot

Reply:
I'm not quite understanding, why can't this all be in the same TX?

Reply:
think it can now, I think this was an issue with the autogenerated int ids causing ordering problems but using the uuids should be fine